packages/rocketchat-lib/lib/RoomTypesCommon.js

```diff
@@ -1,61 +1,67 @@
-/* globals roomExit*/
-this.roomTypesCommon = class {
+/* globals roomExit */
+import { RoomTypeConfig } from './RoomTypeConfig';
+
+export class RoomTypesCommon {
 	constructor() {
 		this.roomTypes = {};
 		this.roomTypesOrder = [];
 		this.mainOrder = 1;
 	}
 
-	/* Adds a room type to app
-	@param identifier An identifier to the room type. If a real room, MUST BE the same of `db.rocketchat_room.t` field, if not, can be null
-	@param order Order number of the type
-	@param config
-	icon: icon class
-	label: i18n label
-	route:
-	name: route name
-	action: route action function
-	identifier: room type identifier
-	*/
-
-	add(identifier = Random.id(), order, config) {
-		if (this.roomTypes[identifier] != null) {
+	/**
+	 * Adds a room type to the application.
+	 *
+	 * @param {RoomTypeConfig} roomConfig
+	 * @returns {void}
+	 */
+	add(roomConfig) {
+		if (!(roomConfig instanceof RoomTypeConfig)) {
+			throw new Error('Invalid Room Configuration object, it must extend "RoomTypeConfig"');
+		}
+
+		if (this.roomTypes[roomConfig.identifier]) {
 			return false;
 		}
-		if (order == null) {
-			order = this.mainOrder + 10;
+
+		if (!roomConfig.order) {
+			roomConfig.order = this.mainOrder + 10;
 			this.mainOrder += 10;
 		}
+
 		this.roomTypesOrder.push({
-			identifier,
-			order
+			identifier: roomConfig.identifier,
+			order: roomConfig.order
 		});
-		this.roomTypes[identifier] = {...config, identifier};
-		if (config.route && config.route.path && config.route.name && config.route.action) {
+
+		this.roomTypes[roomConfig.identifier] = roomConfig;
+
+		if (roomConfig.route && roomConfig.route.path && roomConfig.route.name && roomConfig.route.action) {
 			const routeConfig = {
-				name: config.route.name,
-				action: config.route.action
+				name: roomConfig.route.name,
+				action: roomConfig.route.action
 			};
+
 			if (Meteor.isClient) {
 				routeConfig.triggersExit = [roomExit];
 			}
-			return FlowRouter.route(config.route.path, routeConfig);
+
+			return FlowRouter.route(roomConfig.route.path, routeConfig);
```

Comment:
Just realized, this method say we return void yet here we return whatever `FlowRouter.route` returns. :thinking: 

What are your thoughts @mrsimpson on changing this method?

Reply:
I believe it should rerun void. It’s an anti pattern in some parts of the RC code to just make the last statement of a function return - without meaningful semantics

Reply:
E. G. Create a bunch of settings and the last statement is return this.add()